### PR TITLE
feat(ingest/glossary): add owner type in business glossary connector

### PR DESCRIPTION
**Pull request by AvaniSiddhapuraAPT**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by sgomezvillamor** [comment]
My team was planning to work in this contribution and just found this PR, so this is great!

If I may add a requirement, it would be great if the ownership type could be set with ownership types from enum as well as custom ownership types.

Also, I noted there is some code for the validation that could be replaced with an existing function in the codebase, which by the way already considers custom ownership types :-) 

https://github.com/datahub-project/datahub/blob/11f7804b1ea2f47495c336e8d91d7dc380ccd228/metadata-ingestion/src/datahub/emitter/mce_builder.py#L355

Hope this helps 😄

**Comment by anshbansal** [comment]
> My team was planning to work in this contribution and just found this PR, so this is great!
> 
> If I may add a requirement, it would be great if the ownership type could be set with ownership types from enum as well as custom ownership types.
> 
> Also, I noted there is some code for the validation that could be replaced with an existing function in the codebase, which by the way already considers custom ownership types :-)
> 
> https://github.com/datahub-project/datahub/blob/11f7804b1ea2f47495c336e8d91d7dc380ccd228/metadata-ingestion/src/datahub/emitter/mce_builder.py#L355
> 
> Hope this helps 😄

Hey @sgomezvillamor 
Avani is ramping up on ingestion code and so we want to keep the scope limited for now as we are trying to ramp her up on different items. Please feel free to send a follow up PR. Will try to get this merged in soon & you can send a follow up. Thanks for your understanding.

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py [hunk 1]
```diff
@@ -148,22 +149,41 @@ def make_glossary_term_urn(
 
 def get_owners(owners: Owners) -> models.OwnershipClass:
     owners_meta: List[models.OwnerClass] = []
+
+    valid_owner_types = [
```
Comment (skrydal) [inline metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py:153]: There is already a function implementing that functionality - https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/emitter/mce_builder.py#L347

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py [hunk 2]
```diff
@@ -148,22 +149,41 @@ def make_glossary_term_urn(
...
+        )
+
     if owners.users is not None:
-        owners_meta = owners_meta + [
+        owners_meta += [
             models.OwnerClass(
                 owner=make_user_urn(o),
-                type=models.OwnershipTypeClass.DEVELOPER,
+                type=getattr(
+                    models.OwnershipTypeClass, owners.type or "DEVELOPER"
```
Comment (skrydal) [inline metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py:170]: Is it possible that usage of default value for `ConfigModel` field, like for example [here](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/transformer/add_dataset_ownership.py#L102) could let us avoid writing this logic and repeating it (along with default value)?

**Comment by sgomezvillamor** [comment]
Custom enum type is a requirement for us so sharing this here https://github.com/datahub-project/datahub/pull/9805
Of course, I can also update my branch later if this PR is merged.
Thanks

**Pull request closed without merging**